browser/src/Services/AutoClosingPairs.ts

```diff
@@ -31,15 +32,14 @@ export const activate = (
 
     let subscriptions: Oni.DisposeFunction[] = []
 
-    const handleOpenCharacter = (pair: IAutoClosingPair, editor: Oni.Editor) => () => {
+    const handleOpenCharacter = (
+        pair: IAutoClosingPair,
+        editor: Oni.Editor,
+        openCharacterSameAsClosed: boolean,
+    ) => () => {
         const neovim: NeovimInstance = editor.neovim as any
         neovim.blockInput(async (inputFunc: any) => {
-            // TODO: PERFORMANCE: Look at how to collapse this instead of needed multiple asynchronous calls.
-            await inputFunc(pair.open + pair.close)
-
-            const pos = await neovim.callFunction("getpos", ["."])
-            const [, oneBasedLine, oneBasedColumn] = pos
-            await editor.activeBuffer.setCursorPosition(oneBasedLine - 1, oneBasedColumn - 2)
+            checkOpenCharacter(inputFunc, pair, editor, openCharacterSameAsClosed)
```

Comment:
Looking at the logs from the automation, I do see one bug here - we should either be `return`ing this or `await`'ing it, so that we're actually blocking.

What's happening here is that we're not really blocking the input - so we're allowing the editor to accept further input, which might cause unpredictable races.

In the log I saw from automation, with the resulting string of `window.setTimeout()()`, this could actually explain why we're seeing it. When the first `(` is pressed, we start the logic for `checkOpenCharacter`, but we don't wait for it to complete. In the meantime, a second `(` comes in, and if in the first `checkOpenCharacter`, we haven't placed the cursor, we're going to get unpredictable results. So it seems like there is an actual race condition here that may be impacting the CiTests.